### PR TITLE
Update pathio to use s3 encryption

### PR DESCRIPTION
**Pull request by alsmola**
This change updates `pathio` to a version that support S3 encryption.

For more details, and related changes, see [SEC-139](https://clever.atlassian.net/browse/SEC-139).

Let me know how I might make sure Titanium Transactions uses this latest version.

### CODE REVIEW

File: cmd/oplog-dump/oplogdump.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 	"time"
 
 	"github.com/cenkalti/backoff"
-	"gopkg.in/Clever/pathio.v1"
+	"gopkg.in/Clever/pathio.v3"
```
Comment (natebrennand) [inline cmd/oplog-dump/oplogdump.go]: such a nice clean diff

**Comment by natebrennand** [comment]
LGTM

T_T will be using the latest version once this is deployed in prod. You should coordinate with @d-sparks on that.

**Comment by natebrennand** [comment]
build seems to be breaking due to our breaking API change

**Comment by alsmola** [comment]
Fixed API change.

**Comment by alsmola** [comment]
Actually, see the earlier LGTM, will assume that holds up here.

**Pull request merged**